servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6834,6 +6881,11 @@ public void putTenant(ResourceContext ctx, String providerDomain, String provide
 
         setupTenantAdminPolicyInProvider(ctx, providerDomain, providerService, tenantDomain,
                 false, auditRef, caller);
+
+        // If listed as service provider, register dependency between the provider service and the tenant domain (unless dependency exist
+        // following creation of a previous resource group)
+        String provider = providerDomain + "." + providerService;
```

Comment:
in the same function further up from there we have

`if (!validateTenancyObject(detail, tenantDomain, providerDomain + "." + providerService)) {`

So rather than doing the same providerDomain + "." + providerService; twice, define the value before the validateTenancyObject line and then use it here as well.

We should also make it provider as final String.